### PR TITLE
[DOC] Revisions for call-seq in doc guidelines

### PR DESCRIPTION
**Pull request by BurdetteLamar**
- Splits certain guidelines for singleton and instance method.
- Calls for instance method to not prefix anything (like RDoc itself for a Ruby-coded  instance method);  e.g.:
  -  `count -> integer`, not `array.count`,.
  -  `<=> other -> integer or nil`, not  `hash <=> other -> integer or nil`.
- Groups previous guidelines into Arguments, Block, Return types, Aliases.

### CODE REVIEW

File: doc/contributing/documentation_guide.md
Comment (peterzhu2118) [inline doc/contributing/documentation_guide.md:222]: These code blocks are not valid C, we should just leave them as plain text.

File: doc/contributing/documentation_guide.md
Comment (peterzhu2118) [inline doc/contributing/documentation_guide.md:229]: Do we encourage the form `:call-seq:` or `call-seq:`? They both seem to work but `call-seq:` seems to be much more common.

File: doc/contributing/documentation_guide.md
Comment (peterzhu2118) [inline doc/contributing/documentation_guide.md:231]: We should use `->` rather than the right allow symbol `→`.

File: doc/contributing/documentation_guide.md
Comment (peterzhu2118) [inline doc/contributing/documentation_guide.md:251]:
```suggestion
*    <=> other -> -1, 0, 1, or nil
```

File: doc/contributing/documentation_guide.md
Comment (peterzhu2118) [inline doc/contributing/documentation_guide.md:275]: The description says:

> If the behavior is different with an omitted or an explicit argument

But it looks like the behavior is the same regardless of whether the argument is present or not?

Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md:275]: Poorly chosen example.  What's a better one?

File: doc/contributing/documentation_guide.md
Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md]: Fixed (all instances).

File: doc/contributing/documentation_guide.md
Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md]: Fixed.

File: doc/contributing/documentation_guide.md
Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md]: Fixed.

File: doc/contributing/documentation_guide.md
Comment (peterzhu2118) [reply doc/contributing/documentation_guide.md]: For example, `Enumerable#max` is:

```
 *    max     -> element
 *    max(n)  -> array
```

Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md]: Got it, thanks.

File: doc/contributing/documentation_guide.md
Comment (jeremyevans) [reply doc/contributing/documentation_guide.md:229]: Ruby uses `call-seq:` (1782 cases), not `:call-seq:` (1 case in string.c that should be fixed).

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks mostly good. One requested change, after that, I'm fine to merge if @peterzhu2118 is.

File: doc/contributing/documentation_guide.md
Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md]: Fixed (4 places).

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Pull request merged**
